smile_recognition.py

```diff
@@ -0,0 +1,176 @@
+import torch
+from torch.utils.data import Dataset, DataLoader
+from torch.utils.data.dataset import random_split
+from torchvision.utils import make_grid
+import os
+from torchvision import transforms
+from PIL import Image
+from matplotlib import pyplot as plt
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from tqdm import tqdm
+from sklearn.metrics import confusion_matrix
+from smile_net import Net
+
+torch.manual_seed(2)
+
+# def show(imgs):
+#   grid = make_grid(imgs, nrow=5)
+#   plt.imshow(grid.permute(1, 2, 0))
+#   plt.show()
+
+
+class FaceDataset(Dataset):
+    def __init__(self, img_dir, transform=None):
+        self.img_dir = img_dir
+        self.img_paths = []
+        self.img_labels = []
+        self.transform = transform
+
+        positive_path = os.path.join(img_dir, "positives/positives7")
+        negative_path = os.path.join(img_dir, "negatives/negatives7")
```

Comment:
paths here are hard-coded - maybe have inputs of `positives_path negatives_path`